MapboxCoreNavigation/CLLocation.swift

```diff
@@ -88,25 +88,30 @@ extension CLLocation {
         }
         
         guard let closest = Polyline(nearByCoordinates).closestCoordinate(to: coordinate) else { return nil }
-        guard let calculatedCourseForLocationOnStep = interpolatedCourse(along: nearByCoordinates) else { return nil }
+        guard let projectedLocation = Polyline(nearByCoordinates).trimmed(from: closest.coordinate, distance: projectedDistance).coordinates.last else { return nil }
+        guard let calculatedCourseForLocationOnStep = interpolatedCourse(along: nearByCoordinates, alternateCoordinate: projectedLocation) else { return nil }
         
         let userCourse = calculatedCourseForLocationOnStep
-        let userCoordinate = closest.coordinate
+        let userCoordinate = projectedLocation
         guard let firstCoordinate = legProgress.leg.steps.first?.coordinates?.first else { return nil }
         
         guard shouldSnap(toRouteWith: calculatedCourseForLocationOnStep, distanceToFirstCoordinateOnLeg: self.coordinate.distance(to: firstCoordinate)) else { return nil }
         
         return CLLocation(coordinate: userCoordinate, altitude: altitude, horizontalAccuracy: horizontalAccuracy, verticalAccuracy: verticalAccuracy, course: userCourse, speed: speed, timestamp: timestamp)
     }
     
+    var projectedDistance: CLLocationSpeed {
+        // Account speed being zero.
+        return max(speed * RouteControllerDeadReckoningTimeInterval, 1)
```

Comment:
If the user is stationary, this puts the user 1&nbsp;meter ahead of their actual location. Not a significant problem in practice, but this will throw a wrench into future location-based tests.